### PR TITLE
adding hint method to aggregate

### PR DESCRIPTION
**Pull request by lineus**
MongoDB 3.6 added the hint option to aggregations. you can currently add a hint with the .option method, but having a convenience method like the one for allowDiskUse makes sense to me.

I also updated the helper method formerly known as onlyTestMongo34. I made the version it looks for variable and changed the name to reflect the new behavior.

Test expects that the resulting array of docs is sorted according to the hint.

```
mongoose>: npm test
....
  2 failing

  1) Model
       bug fixes
         3.6 features
           watch() (gh-5964):
     Uncaught MongoError: Majority read concern requested, but it is not supported by the storage engine.
      at queryCallback (node_modules/mongodb-core/lib/cursor.js:223:25)
      at node_modules/mongodb-core/lib/connection/pool.js:541:18
      at _combinedTickCallback (internal/process/next_tick.js:131:7)
      at process._tickCallback (internal/process/next_tick.js:180:9)

  2) Model
       bug fixes
         3.6 features
           watch() before connecting (gh-5964):
     Uncaught MongoError: Majority read concern requested, but it is not supported by the storage engine.
      at queryCallback (node_modules/mongodb-core/lib/cursor.js:223:25)
      at node_modules/mongodb-core/lib/connection/pool.js:541:18
      at _combinedTickCallback (internal/process/next_tick.js:131:7)
      at process._tickCallback (internal/process/next_tick.js:180:9)



npm ERR! Test failed.  See above for more details.
```
I'm assuming it's my env that's causing these failures.

### CODE REVIEW

File: test/aggregate.test.js [hunk 1]
```diff
@@ -48,20 +48,27 @@ function setupData(db, callback) {
...
+function onlyTestAtOrAbove(semver, ctx, done) {
   start.mongodVersion(function(err, version) {
     if (err) {
       done(err);
       return;
     }
-    var mongo34 = version[0] > 3 || (version[0] === 3 && version[1] >= 4);
-    if (!mongo34) {
+
+    var [major, minor] = semver.split('.').map(function(s) {
```
Comment (vkarpov15) [inline test/aggregate.test.js]: Array destructuring doesn't work in node v4 and v5, which is one of the reason why tests are failing

Comment (lineus) [reply test/aggregate.test.js]: ok, I'll fix it.

**Comment by lineus** [comment]
The destructuring error is gone, but now on node 4, 6, and 7 there are different errors. Not just different from before, but different from each other, and in places that I can't readily explain by the changes I made. I'll keep digging into it. I know I shouldn't *want* to re-run the tests on the same commit in travis, it feels insane, but I do :)

**Comment by lineus** [comment]
I created a new branch and a new PR #6272 with the same changes to test this code again, all tests passed. Is there a better strategy than this when it seems something went wrong on travis CI?

**Pull request closed without merging**